src/mongo/shell/shell_utils_launcher.cpp

```diff
@@ -304,7 +304,7 @@ ProgramRunner::ProgramRunner(const BSONObj& args, const BSONObj& env) {
         if (str == "--port") {
             _port = -2;
         } else if (_port == -2) {
-            _port = strtol(str.c_str(), 0, 10);
+            parseNumberFromString<int>(str.c_str(), &_port);
```

Comment:
In the event that str does not parse as a number, the old behavior set _port to 0, and this will leave it unchanged at -2.

However, this function uses verify(), above, which means that it also sometimes throws exceptions on errors. It would be OK here to wrap this parseNumberFromString call in uassertStatusOK() to get a bad status converted into a thrown exception:

uassertStatusOK(parseNumberFromString<int>(str, &_port));

I mentioned this on another change in this patch. Also, you don't have to call c_str(), as parseNumberFromString can accept std::strings directly.